### ISSUE TITLE
when i try to edit a domain in the client area i get this error

### ISSUE THREAD

**Issue by solucionesuno** [openprovider/OP-WHMCS7#48]
`Error: Call to undefined method WeDevelopCoffee\wPower\Domain\AdditionalFields::setRegistrarAdditionalFields() in /var/www/vhosts/iontics.com/httpdocs/panel/modules/registrars/openprovider/Library/AdditionalFields.php:90
Stack trace:
#0 /var/www/vhosts/iontics.com/httpdocs/panel/modules/registrars/openprovider/helpers.php(21): OpenProvider\WhmcsRegistrar\Library\AdditionalFields->get()
#1 /var/www/vhosts/iontics.com/httpdocs/panel/resources/domains/additionalfields.php(6): openprovider_additional_fields()
#2 /var/www/vhosts/iontics.com/httpdocs/panel/vendor/whmcs/whmcs-foundation/lib/Domains/AdditionalFields.php(0): unknown()
#3 /var/www/vhosts/iontics.com/httpdocs/panel/vendor/whmcs/whmcs-foundation/lib/Domains/AdditionalFields.php(0): WHMCS\Domains\AdditionalFields->fetch('additionalfield...')
#4 /var/www/vhosts/iontics.com/httpdocs/panel/vendor/whmcs/whmcs-foundation/lib/Domains/AdditionalFields.php(0): WHMCS\Domains\AdditionalFields->loadFieldsData()
#5 /var/www/vhosts/iontics.com/httpdocs/panel/vendor/whmcs/whmcs-foundation/lib/Domains/AdditionalFields.php(0): WHMCS\Domains\AdditionalFields->setTLD('.com')
#6 /var/www/vhosts/iontics.com/httpdocs/panel/admin/clientsdomains.php(0): WHMCS\Domains\AdditionalFields->setDomain('menjadorscollec...')
#7 {main}`

**Comment by hostingwalk** [comment]
I can confirm this bug.

I fixing this temporarily with uncommented the below line with # in the file:  /modules/registrars/openprovider/Library/AdditionalFields.php

```
 // Set the additional fields of OpenProvider
 #$this->WAdditionalFields->setRegistrarAdditionalFields($additionaldomainfields);
```

Hope there is coming a solution for this bug.

**Comment by tamerciaga** [comment]
We have this same issue, why is this happening? Uncommenting the line is not a real solution.

**Comment by soberoctober** [comment]
@hostingwalk does this happen with a .com TLD? Which additional fields do you have assigned for it?

**Comment by solucionesuno** [comment]
I find finally that its happen when you activate the addon of openprovider.

disable addon problem is solved. so something is worng with the addon

**Comment by soberoctober** [comment]
fixed in [v3.0](https://github.com/openprovider/OP-WHMCS7/tree/v3.0)
